src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentracing.contrib.tracerresolver;
+
+import io.opentracing.Tracer;
+
+import java.util.ServiceLoader;
+
+/**
+ * {@code TracerResolver} API definition looks for one or more registered {@link TracerResolver} implementations
+ * using the {@link ServiceLoader}.
+ * <p>
+ * If no {@link TracerResolver} implementations are found, the {@link #resolveTracer()} method will fallback to
+ * {@link ServiceLoader} lookup of the {@link Tracer} service itself.
+ */
+public abstract class TracerResolver {
+    private static final ServiceLoader<TracerResolver> RESOLVERS = ServiceLoader.load(TracerResolver.class);
+    private static final ServiceLoader<Tracer> FALLBACK = ServiceLoader.load(Tracer.class);
+
+    private static Tracer resolved = null; // [ST] I'm still in doubt whether the resolved result should be cached or not!
```

Comment:
The resolver should ideally only be called once, when registering against the GlobalTracer, so probably no need.